go/libraries/doltcore/sqle/enginetest/dolt_queries_schema_override.go

```diff
@@ -0,0 +1,1034 @@
+			},
+			{
+				Query:    "SET @@dolt_schema_override_commit=@commit1;",
+				Expected: []sql.Row{{}},
+			},
+			{
+				Query:    "SELECT * from deletedTable;",
+				Expected: []sql.Row{},
+			},
+		},
+	},
+	{
+		Name: "Table Existence: Table exists in the pinned schema commit, NOT in AS OF data commit",
+		SetUpScript: []string{
+			"create table deletedTable (pk int primary key, c1 varchar(255));",
+			"insert into deletedTable values (1, 'one');",
+			"call dolt_commit('-Am', 'adding table deletedTable on main');",
+			"SET @commit1 = hashof('HEAD');",
+
+			"drop table deletedTable;",
+			"call dolt_commit('-Am', 'deleting table deletedTable on main');",
+			"SET @commit2 = hashof('HEAD');",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:          "SELECT * from deletedTable;",
+				ExpectedErrStr: "table not found: deletedtable",
+			},
+			{
+				Query:    "SET @@dolt_schema_override_commit=@commit1;",
+				Expected: []sql.Row{{}},
+			},
+			{
+				Query:    "SELECT * from deletedTable AS OF @commit2;",
+				Expected: []sql.Row{},
+			},
+		},
+	},
+
+	// INDEXES
+	{
+		Name: "Indexes: Index in the pinned schema, but not in the data commit",
+		SetUpScript: []string{
+			"create table t (pk int primary key, c1 varchar(255), key c1_idx(c1));",
+			"insert into t values (1, 'one');",
+			"call dolt_commit('-Am', 'adding table t with index on main');",
+			"SET @commit1 = hashof('HEAD');",
+
+			"alter table t drop index c1_idx;",
+			"update t set c1='two';",
+			"call dolt_commit('-Am', 'adding table t with index on main');",
+			"SET @commit2 = hashof('HEAD');",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				// Going back to @commit1 with AS OF should use the available index
+				Query:           "SELECT c1 from t as of @commit1 where c1 > 'o';",
+				Expected:        []sql.Row{{"one"}},
+				ExpectedIndexes: []string{"c1_idx"},
+			},
+			{
+				// The tip of HEAD does not have an index
+				Query:           "SELECT c1 from t where c1 > 'o';",
+				Expected:        []sql.Row{{"two"}},
+				ExpectedIndexes: []string{},
+			},
+			{
+				// Set the overridden schema to the point where an index existed
+				Query:    "SET @@dolt_schema_override_commit=@commit1;",
+				Expected: []sql.Row{{}},
+			},
+			{
+				// Using the overridden index, we should still get the latest data, but without using the index
+				Query:           "SELECT c1 from t where c1 > 'o';",
+				Expected:        []sql.Row{{"two"}},
+				ExpectedIndexes: []string{},
+			},
+			{
+				// Set the overridden schema to the point where an index existed
+				Query:    "SET @@dolt_schema_override_commit=@commit2;",
+				Expected: []sql.Row{{}},
+			},
+			{
+				// Going back to @commit1 for data, but using @commit2 for schema
+				Query:           "SELECT c1 from t as of @commit1 where c1 > 'o';",
+				Expected:        []sql.Row{{"one"}},
+				ExpectedIndexes: []string{"c1_idx"},
+			},
+		},
+	},
+
+	// AS OF TEST CASES
```

Comment:
Other forms you're missing that could use some coverage:

```
`database/branch`.`table`
`database/HEAD~`.`table`
`database/tag`.`table`
`database/commitHash`.`table`
```

Reply:
Good call – tag revision and commit hash revision syntax triggered a panic on a bad type cast. Fixed and added tests.